NiL.JS/Core/Tools.cs

```diff
@@ -1887,5 +1887,340 @@ internal static LambdaExpression BuildJsCallTree(string name, Expression functio
+                        }
+                        if (a._fields != null)
+                        {
+                            for (var e = a._fields.GetEnumerator(); e.MoveNext();)
+                            {
+                                if (i++ > 0)
+                                    s.Append(", ");
+                                s.Append(e.Current.Key).Append(": ");
+                                s.Append(Tools.JSValueToObjectString(e.Current.Value, maxRecursionDepth, recursionDepth + 1));
+                            }
+                        }
+                        s.Append(" ]");
+
+                        return s.ToString();
+                    }
+                    else
+                    {
+                        string typeName = Tools.GetTypeName(v);
+                        if (recursionDepth >= maxRecursionDepth)
+                            return typeName;
+
+                        StringBuilder s = new StringBuilder(typeName);
+                        s.Append(" { ");
+
+                        JSObject o = v as JSObject;
+                        if (o == null)
+                            o = v._oValue as JSObject;
+
+                        
+                        int i = 0;
+                        for (var e = o.GetEnumerator(true, EnumerationMode.RequireValues); e.MoveNext();)
+                        {
+                            if (i++ > 0)
+                                s.Append(", ");
+                            s.Append(e.Current.Key).Append(": ");
+                            s.Append(Tools.JSValueToObjectString(e.Current.Value, maxRecursionDepth, recursionDepth + 1));
+                        }
+
+                        s.Append(" }");
+
+                        return s.ToString();
+                    }
+            }
+            return v.ToString();
+        }
+        
+        public static string JSValueToString(JSValue v)
+        {
+            if (v == null)
+                return "null";
+
+            if (v.ValueType == JSValueType.Object)
+            {
+                if (v._oValue == null)
+                    return "null";
+
+                var o = v as ObjectWrapper;
+                if (o == null)
+                    o = v._oValue as ObjectWrapper;
+                if (o != null)
+                    return o.Value.ToString();
+            }
+
+            return v.ToString();
+        }
+
+
+        public static string FormatArgs(Arguments args)
+        {
+            return FormatArgs(args, 0, args.length, true);
+        }
+        
+        public static string FormatArgs(Arguments args, int start)
+        {
+            return FormatArgs(args, start, args.length, true);
+        }
+        
+        public static string FormatArgs(Arguments args, int start, int stop)
+        {
+            return FormatArgs(args, start, stop, true);
+        }
+
+        public static string FormatArgs(Arguments args, int start, int stop, bool formatted)
```

Comment:
I not found any real use this variant of function. I think `start`, `stop` and `formatted` are very specific parameters.